src/components/dialogs/filter/criteria-based/filter-free-properties.tsx

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import {
+    ErrorInput,
+    FieldErrorAlert,
+    useSnackMessage,
+} from '@gridsuite/commons-ui';
+import { Button, Grid, ListItem } from '@mui/material';
+import {
+    CRITERIA_BASED,
+    EQUIPMENT_TYPE,
+} from 'components/utils/field-constants';
+import { useFieldArray, useWatch } from 'react-hook-form';
+import { FormattedMessage } from 'react-intl';
+import { Hvdc, Line } from 'utils/equipment-types';
+import {
+    FREE_FILTER_PROPERTIES,
+    SUBSTATION_FILTER_PROPERTIES,
+} from './filter-properties';
+import FilterProperty, {
+    PROPERTY_NAME,
+    PROPERTY_VALUES,
+    PROPERTY_VALUES_1,
+    PROPERTY_VALUES_2,
+} from './filter-property';
+import AddIcon from '@mui/icons-material/Add';
+import { useEffect, useMemo, useState } from 'react';
+import { fetchAppsAndUrls } from 'utils/rest-api';
+
+function fetchPredefinedProperties() {
+    return fetchAppsAndUrls().then((res) => {
+        const studyMetadata = res.find(
+            (metadata: any) => metadata.name === 'Study'
+        );
+        if (!studyMetadata) {
+            return Promise.reject(
+                'Study entry could not be found in metadatas'
+            );
+        }
+
+        return studyMetadata?.predefinedEquipmentProperties?.substation;
+    });
+}
+
+interface FilterFreePropertiesProps {
+    freePropertiesType: string;
```

Comment:
this component is designed to handle two types of freeProperties (FREE_FILTER_PROPERTIES and SUBSTATION_FILTER_PROPERTIES) not any string

Reply:
I've replaced it by an enum